Include/internal/pycore_gc.h

```diff
@@ -20,7 +20,6 @@ typedef struct {
 } PyGC_Head;
 
 #define _Py_AS_GC(o) ((PyGC_Head *)(o)-1)
-#define _PyGC_Head_UNUSED PyGC_Head
```

Comment:
There are probably other containers that we will statically allocate (and drop from GC).  That's why I added `_PyGC_Head_UNUSED`.  Would it make sense use a similar approach (negative size) for each of those cases?

Reply:
Maybe, but I am not sure. It's case-by-case.
But for now, only tuple uses this macro. If we remove the GC header from constant tuple, this macro should be removed too.
When we need it again, we can add it again.